lnrpc/websocket_proxy.go

```diff
@@ -129,6 +134,11 @@ func (p *WebsocketProxy) upgradeToWebSocketProxy(w http.ResponseWriter,
 		request.Method = m
 	}
 
+	// Is this a call to a client-streaming RPC method?
+	streamParam := r.URL.Query().Get(ClientStreamingParam)
```

Comment:
is there really no way of extracting what RPC is being called? Then we could've easily mapped that to whether it was client streaming or not.

Reply:
Unfortunately not, as far as I can tell. I even tried to use some black reflection magic but that also got me nowhere. I even checked if upgrading to the latest version of `grpc-gateway` somehow exposed more meta information that we could use for the lookup but that doesn't seem to be the case either. We could fork the `grpc-gateway` library but that seems excessive for (currently) just two request-streaming RPCs.

Reply:
So what is set in the `r.URL` path? Is it blank?

Reply:
It is set to the full URL (for example `/v1/channels/acceptor`). But we don't have any reverse mapping to the RPC method that's being called for it (except in `lnrpc/rest-annotations.yaml` which is not compiled into the binary). There's only the generated code in the `*.pb.gw.go` files and there's no mapping metadata there either (each HTTP method is generated individually instead of using a generic mapping).
We could of course keep an extra hard coded list of client-streaming RPCs somewhere and match against that. But then we'd very likely forget to add new URLs once we add new RPCs.

Reply:
Discussed this offline and decided to go with a hard coded list maintained in `lnrpc` instead of putting the burden of the additional request parameter onto the client.